JFRTest#jfrPerfTest fails with Quarkus `main`

## How to reproduce

```
git clone https://github.com/quarkusio/quarkus.git
cd quarkus
./mvnw -Dquickly
cd ../
git clone https://github.com/Karm/mandrel-integration-tests
cd mandrel-integration-tests
mvn clean verify -Dquarkus.version=999-SNAPSHOT -Ptestsuite -Dtest=JFRTest#jfrPerfTest
```

## Failure:

```
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 78.177 s <<< FAILURE! - in org.graalvm.tests.integration.JFRTest
[ERROR] jfrPerfTest{TestInfo}  Time elapsed: 77.691 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: Timeout 10s was reached. Empty webpage does not contain string: `hello' ==> expected: <true> but was: <false>
	at org.graalvm.tests.integration.JFRTest.runBenchmarkForApp(JFRTest.java:355)
	at org.graalvm.tests.integration.JFRTest.startComparisonForBenchmark(JFRTest.java:254)
	at org.graalvm.tests.integration.JFRTest.jfrPerfTestRun(JFRTest.java:214)
	at org.graalvm.tests.integration.JFRTest.jfrPerfTest(JFRTest.java:176)
```

## Additional info

The regression is not caused by a GraalVM change, building with Mandrel 23.1.1.0 also doesn't work.
Setting `quarkus.version` to `3.6.0` makes the test fail in a different stage (possibly a local issue?)

FTR The stack trace in Quarkus leading to the failure is:

```
2023-12-09 00:36:04,630 TRACE [io.ver.cor.htt.imp.HttpServerImpl] (vert.x-eventloop-thread-0) Connection failure: java.lang.Error: java.lang.NullPointerException
	at io.netty.util.internal.PlatformDependent0.newDirectBuffer(PlatformDependent0.java:620)
	at io.netty.util.internal.PlatformDependent0.allocateDirectNoCleaner(PlatformDependent0.java:579)
	at io.netty.util.internal.PlatformDependent.allocateDirectNoCleaner(PlatformDependent.java:774)
	at io.netty.buffer.UnpooledUnsafeNoCleanerDirectByteBuf.allocateDirect(UnpooledUnsafeNoCleanerDirectByteBuf.java:30)
	at io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeNoCleanerDirectByteBuf.allocateDirect(UnpooledByteBufAllocator.java:186)
	at io.netty.buffer.UnpooledDirectByteBuf.<init>(UnpooledDirectByteBuf.java:64)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.<init>(UnpooledUnsafeDirectByteBuf.java:41)
	at io.netty.buffer.UnpooledUnsafeNoCleanerDirectByteBuf.<init>(UnpooledUnsafeNoCleanerDirectByteBuf.java:25)
	at io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeNoCleanerDirectByteBuf.<init>(UnpooledByteBufAllocator.java:181)
	at io.netty.buffer.UnpooledByteBufAllocator.newDirectBuffer(UnpooledByteBufAllocator.java:91)
	at io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:188)
	at io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:179)
	at io.netty.buffer.Unpooled.directBuffer(Unpooled.java:128)
	at io.netty.handler.codec.http.HttpObjectEncoder.<clinit>(HttpObjectEncoder.java:59)
	at java.base@21.0.1/java.lang.Class.ensureInitialized(DynamicHub.java:595)
	at java.base@21.0.1/java.lang.Class.ensureInitialized(DynamicHub.java:595)
	at io.vertx.core.http.impl.HttpServerWorker.configureHttp1OrH2C(HttpServerWorker.java:276)
	at io.vertx.core.http.impl.HttpServerWorker.handleHttp1(HttpServerWorker.java:209)
	at io.vertx.core.http.impl.HttpServerWorker.access$200(HttpServerWorker.java:50)
	at io.vertx.core.http.impl.HttpServerWorker$1.configure(HttpServerWorker.java:180)
	at io.vertx.core.http.impl.Http1xOrH2CHandler.end(Http1xOrH2CHandler.java:59)
	at io.vertx.core.http.impl.Http1xOrH2CHandler.channelRead(Http1xOrH2CHandler.java:38)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base@21.0.1/java.lang.Thread.runWith(Thread.java:1596)
	at java.base@21.0.1/java.lang.Thread.run(Thread.java:1583)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:211)
Caused by: java.lang.NullPointerException
	at io.netty.util.internal.PlatformDependent0.newDirectBuffer(PlatformDependent0.java:614)
	... 44 more
```

Which is the result of `PlatformDependent.USE_DIRECT_BUFFER_NO_CLEANER` being set (and observed by other classes) at build time while `PlatformDependent0.DIRECT_BUFFER_CONSTRUCTOR` is null at run time (in Quarkus, because in my tests above it appears to be set :/)

```
DEBUG [io.net.uti.int.PlatformDependent0] (Thread-3) java.nio.DirectByteBuffer.<init>(long, {int,long}): unavailable
```

--- a/extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyProcessor.java
+++ b/extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyProcessor.java
@@ -162,8 +162,28 @@ class NettyProcessor {
             log.debug("Not registering Netty native kqueue classes as they were not found");
         }
 
-        builder.addRuntimeReinitializedClass("io.netty.util.internal.PlatformDependent");
-        builder.addRuntimeReinitializedClass("io.netty.util.internal.PlatformDependent0");
+        builder.addRuntimeReinitializedClass("io.netty.util.internal.PlatformDependent")
+                .addRuntimeReinitializedClass("io.netty.util.internal.PlatformDependent0")
+                .addRuntimeReinitializedClass("io.netty.buffer.PooledByteBufAllocator");
+
+        if (QuarkusClassLoader.isClassPresentAtRuntime("io.netty.buffer.UnpooledByteBufAllocator")) {
+            builder.addRuntimeReinitializedClass("io.netty.buffer.UnpooledByteBufAllocator")
+                    .addRuntimeReinitializedClass("io.netty.buffer.Unpooled")
+                    .addRuntimeReinitializedClass("io.vertx.core.http.impl.Http1xServerResponse")
+                    .addRuntimeReinitializedClass("io.netty.handler.codec.http.HttpObjectAggregator")
+                    .addRuntimeReinitializedClass("io.netty.handler.codec.ReplayingDecoderByteBuf")
+                    .addRuntimeReinitializedClass("io.vertx.core.parsetools.impl.RecordParserImpl");
+
+            if (QuarkusClassLoader.isClassPresentAtRuntime("io.vertx.ext.web.client.impl.MultipartFormUpload")) {
+                builder.addRuntimeReinitializedClass("io.vertx.ext.web.client.impl.MultipartFormUpload");
+            }
+
+            if (QuarkusClassLoader
+                    .isClassPresentAtRuntime("org.jboss.resteasy.reactive.client.impl.multipart.QuarkusMultipartFormUpload")) {
+                builder.addRuntimeReinitializedClass(
+                        "org.jboss.resteasy.reactive.client.impl.multipart.QuarkusMultipartFormUpload");
+            }
+        }
 
         return builder //TODO: make configurable
                 .build();

